finance_manager/src/main/java/com/manager/finance/mapstruct/mapper/TransactionMapper.java

```diff
@@ -1,26 +1,39 @@
 package com.manager.finance.mapstruct.mapper;
 
 import com.manager.finance.transaction.model.entity.Transaction;
+import com.sandbox.model.CategoryResponseDto;
 import com.sandbox.model.TransactionRequestDto;
 import com.sandbox.model.TransactionResponseDto;
+import com.sandbox.model.TransactionTypeParameter;
 import java.util.Formatter;
 import java.util.List;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
 import org.mapstruct.Named;
 
-@Mapper(componentModel = "spring")
+@Mapper(componentModel = "spring", uses = {CategoryMapper.class})
+
 public interface TransactionMapper {
 
     @Named("stringDoubleToInt")
-    public static int intToStringDouble(String amount) {
+    public static Integer stringDoubleToIntSource(String amount) {
         return (int) Math.round(Double.parseDouble(amount) * 100);
     }
 
     @Named("intToStringDouble")
-    public static String stringDoubleToInt(int amount) {
-        return new Formatter().format("%.2f",
-                ((double) (amount)) / 100).toString().replace(",", ".");
+    public static String intToStringDouble(Integer amount) {
+        return amount == null ? "0.00"
+                : new Formatter().format("%.2f", ((double) (amount)) / 100).toString();
+    }
+
+    @Named("createCategoryByName")
+    public static CategoryResponseDto stringDoubleToInt(String name) {
```

Comment:
I don't understand what is the purpose of this method. Pls come up with better naming